libraries/Microsoft.Bot.Connector/Authentication/DelegatingCredentialProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// This is just an internal class to allow pre-existing implementation of the request validation to be used with a IServiceClientCredentialFactory.
+    /// </summary>
+    internal class DelegatingCredentialProvider : ICredentialProvider
+    {
+        private IServiceClientCredentialsFactory _credentialFactory;
+
+        public DelegatingCredentialProvider(IServiceClientCredentialsFactory credentialFactory)
+        {
+            _credentialFactory = credentialFactory;
```

Comment:
```suggestion
            _credentialFactory = credentialFactory ?? throw new ArgumentNullException(nameof(credentialFactory));
```

Reply:
fixed - see previous comments - hoping this code doesn't actually ship - as in the ideal world the parameterized version of this code should supersede _before_ we ship 4.11.